### PR TITLE
ci: add make goal for e2e CLI tests

### PR DESCRIPTION
**Pull request by antoineco**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

Facilitate running e2e CLI tests, both locally and in the CI environment, by adding a `make e2e-test` goal.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -128,6 +128,11 @@ test: GOTEST_PKGS := $(foreach pkg,$(filter-out $(GOTEST_EXCLUDE),$(wildcard */)
 test: ## Run unit tests.
 	$(GO) run github.com/onsi/ginkgo/v2/ginkgo -v -p -randomize-all --tags static $(GOTEST_PKGS)
 
+
+.PHONY: e2e-test
+e2e-test: $(BIN) ## Run CLI end-to-end tests.
+	$(GO) run github.com/onsi/ginkgo/v2/ginkgo -v -p -randomize-all test/e2e/cli/...
+
```
Comment (nderjung) [inline Makefile]: I would use `test-*` (as opposed to `*-test`) as the prefix to keep things consistent amongst existing target names (i.e. `buildenvs-*`) and I would create a global target to test both units and e2e or to allow calling each individually:

```diff
+ test: test-unit
+ test: test-e2e
+ test: ## Run all tests.
[...]
- test: ## Run unit tests.
+ test-unit: ## Run unit tests.
[...]
- e2e-test: $(BIN) ## Run CLI end-to-end tests.
+ test-e2e: $(BIN) ## Run CLI end-to-end tests.
```

Comment (antoineco) [reply Makefile]: - Renaming done in [fd436233..efa95e3b](https://github.com/unikraft/kraftkit/compare/fd436233cc697f83f06dcf3fc86a813e996c45d1..efa95e3b4c47bf5ced602c4dc8a0adf68b34bb16)
- Global goal in 9b8c2457

Comment (nderjung) [reply Makefile]: `make help` no longer shows the `test` target:
```
KraftKit developer build targets.

USAGE
  make [VAR=... [VAR=...]] TARGET

TARGETS
  devenv                   Start the development environment container.
  tidy                     Tidy import Go modules.
  fmt                      Format all files according to linting preferences.
  cicheck                  Run CI checks.
  test-unit                Run unit tests.
  test-e2e                 Run CLI end-to-end tests.
  install-golangci-lint    Install the Golang CI lint tool
  properclean              Completely clean the repository's build artifacts.
  help                     Show this help menu and exit.
  buildenv-base            OCI image used for building Unikraft unikernels with kraft.
  buildenv-gcc             OCI image containing a Unikraft-centric build of gcc.
  buildenv-myself-full     OCI image containing the build environment for KraftKit.
  buildenv-myself          OCI image containing KraftKit binaries.
  buildenv-qemu            OCI image containing a Unikraft-centric build of QEMU.
  kraft                    The kraft binary.
  ```

Comment (antoineco) [reply Makefile]: I forgot one hash mark apparently, nice catch!
Fixed in [9b8c2457..bd2f84ef](https://github.com/unikraft/kraftkit/compare/9b8c2457817f06190eed09d918bf5f70e46229eb..bd2f84ef63bac073d587184ec2f1294715231021).

File: .github/workflows/gotests.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ jobs:
         run: make git2go
 
       - name: Run tests
-        run: make test
+        run: make test-unit
```
Comment (nderjung) [inline .github/workflows/gotests.yaml:60]: Just saw this now, to help with PR check clarity, let's rename `Run tests` to `Run unit tests`:

```diff
- name: Run tests
+ name: Run unit tests
```

File: .github/workflows/gotests.yaml [hunk 3]
```diff
@@ -64,19 +64,11 @@ jobs:
...
-
-      - name: Install Ginkgo
-        run: go install github.com/onsi/ginkgo/v2/ginkgo
-
-      - name: Install kraft
-        run: make kraft DOCKER= DISTDIR="$(go env GOPATH)"/bin
-
       - name: Run unit tests
-        run: ginkgo -v -p -randomize-all ./test/e2e/framework/...
+        run: go run github.com/onsi/ginkgo/v2/ginkgo -v -p -randomize-all ./test/e2e/framework/...
```
Comment (nderjung) [inline .github/workflows/gotests.yaml]: I am a bit confused about this step actually, how does it differ from `Run e2e tests`?

Comment (antoineco) [reply .github/workflows/gotests.yaml]: Those contain unit tests for the test framework itself.

The CLI tests are in test/cli.

I'm running the unit tests before the CLI tests to ensure that the framework's code is working as expected. E.g. if we make changes to a matcher and that marcher broke, surely the CLI tests will report erroneous results.

Comment (nderjung) [reply .github/workflows/gotests.yaml]: Gotcha. Your latest update helped clarify things. Thanks!

File: .github/workflows/gotests.yaml (hunk 2 above)
Comment (antoineco) [reply .github/workflows/gotests.yaml:60]: Did some changes to the wording in d83b3274e105b9c7e77eb7043afa021cf504d9f5:

image
image

Feel free to fixup and force-push.

**Comment by antoineco** [comment]
Pushed a quick fixup in [1a52b4a1..974c4d48](https://github.com/unikraft/kraftkit/compare/1a52b4a1da463eb048afe9d3e669755dd02d0f34..974c4d48636f6006b7bfa2892c6f71544f06ebf0). git2go will be installed by make automatically.

**Comment by antoineco** [comment]
@nderjung friendly bump

**Comment by craciunoiuc** [comment]
@nderjung unfriendly bump 👊 

(from my side the changes look okay)

**Review by nderjung — APPROVED** [review]
[Looks good to me, Thanks!](https://xkcd.com/149/)

Approved-by: Alexander Jung <alex@unikraft.io>

**Pull request merged**